### PR TITLE
Dev 1.0.0 datasource

### PR DESCRIPTION
**Pull request by Adamyuanyuan**
In order to make it more convenient for users to manage data sources, we built a unified datasource module with the help of WeBank.These functions are included in the module：
1. DataSourceManager module.  DataSourceManager is the core module of data sources, which is used to manage all kinds of data sources in the big data platform, mainly including the addition, deletion, modification and release of data sources.Version management and other  functions;  #723 
2.  MetaManager module is used for docking metadata of various data sources in the big data platform, mainly including data source connection testing, metadata query and other functions; #724 
3. The DataSource Client module contains the call logic of the Client side, so that users can operate data sources and obtain relevant metadata in the way of the Client side; #725 
4. Added Kafka data source module, its role is to unify the management of Kafka data sources, Kafka is usually used for streaming computing applications; #726

### CODE REVIEW

**Comment by xutaoustc** [comment]
datasource client

File: linkis-public-enhancements/linkis-datasource/datasource-client/src/main/scala/com/webank/wedatasphere/linkis/datasource/client/response/MetadataGetPartitionsResult.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.webank.wedatasphere.linkis.httpclient.dws.annotation.DWSHttpMessageResult
+import com.webank.wedatasphere.linkis.httpclient.dws.response.DWSResult
+import com.webank.wedatasphere.linkis.metadatamanager.common.domain.MetaPartitionInfo
+
+import scala.beans.BeanProperty
+
+@DWSHttpMessageResult("/api/rest_j/v\\d+/metadata/partitions/(\\S+)/db/(\\S+)/table/(\\S+)")
+class MetadataGetPartitionsResult extends DWSResult{
+  @BeanProperty var props: MetaPartitionInfo = _
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/datasource-client/src/main/scala/com/webank/wedatasphere/linkis/datasource/client/response/MetadataGetPartitionsResult.scala:11]: Linkis http-client architecture cannot directly parse a Map to a object.

File: linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+//
+//    val getInfoByDataSourceIdResult = dataSourceClient.getInfoByDataSourceId(
+//        GetInfoByDataSourceIdAction.builder().setDataSourceId(1).setSystem("xx").setUser("hadoop").build()
+//    ).getDataSource
+//
+//    val queryDataSourceResult = dataSourceClient.queryDataSource(QueryDataSourceAction.builder()
+//                                      .setSystem("")
+//                                      .setName("mysql")
+//                                      .setTypeId(1)
+//                                      .setIdentifies("")
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala:48]: Please help delete the comment code.

File: linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import com.webank.wedatasphere.linkis.datasource.client.request._
+import com.webank.wedatasphere.linkis.httpclient.dws.authentication.StaticAuthenticationStrategy
+import com.webank.wedatasphere.linkis.httpclient.dws.config.DWSClientConfigBuilder
+
+import java.util.concurrent.TimeUnit
+
+object TestDataSourceClient {
+  def main(args: Array[String]): Unit = {
+    val clientConfig = DWSClientConfigBuilder.newBuilder()
+      .addServerUrl("http://192.168.10.201:9001")
```
Comment (peacewong) [inline linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala:13]: ip should be modified to 127.0.0.1

File: db/linkis_dml.sql [hunk 4]
```diff
@@ -316,3 +316,30 @@ INSERT INTO `linkis_configuration_config_value` (`configkey_id`, `config_value`,
...
+INSERT INTO `linkis_datasource_type_key` VALUES (4, 1, 'username', '用户名', NULL, 'TEXT', NULL, 1, '用户名', '^[A-Za-z]+$', NULL, NULL, NULL, '2021-04-12 01:54:39', '2021-04-12 01:54:39');
+INSERT INTO `linkis_datasource_type_key` VALUES (5, 1, 'password', '密码', NULL, 'PASSWORD', NULL, 1, '密码', '^[a-zA-Z0-9]{6,16}$', NULL, NULL, NULL, '2021-04-12 01:54:39', '2021-04-12 01:54:39');
+INSERT INTO `linkis_datasource_type_key` VALUES (6, 2, 'file', 'keytab', NULL, 'FILE', NULL, 0, 'keytab文件', NULL, NULL, NULL, NULL, '2021-04-12 02:38:11', '2021-04-12 02:38:11');
+INSERT INTO `linkis_datasource_type_key` VALUES (7, 2, 'envId', '集群环境', '', 'SELECT', NULL, 1, '集群配置', NULL, NULL, NULL, 'http://localhost:8080/api/rest_j/v1/data_source/env_list/all/type/2', '2021-04-13 01:53:57', '2021-04-13 01:53:57');
+
+
+-- ----------------------------
+-- Records of linkis_datasource_type
+-- ----------------------------
+INSERT INTO `linkis_datasource_type` VALUES (1, 'mysql', 'mysql数据库', 'mysql数据库', '关系型数据库', 'https://img.alicdn.com/imgextra/i4/O1CN01uLYwgg1zS93Aq9W8C_!!6000000006712-2-tps-280-176.png', 3);
```
[the remark this answers is not in the document]
Comment (Adamyuanyuan) [reply db/linkis_dml.sql:343]: Thankis. I'll fix it right away.

File: linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java [hunk 5]
```diff
@@ -64,227 +67,415 @@
...
+        return RestfulApiHelper.doAndResponse(() -> {
+                    DataSourceParamKeyDefinition.Scope scope = DataSourceParamKeyDefinition.Scope.valueOf(scopeValue.toUpperCase());
+                    List<DataSourceParamKeyDefinition> keyDefinitions = dataSourceRelateService
+                            .getKeyDefinitionsByType(dataSourceTypeId, scope);
+                    return Message.ok().data("key_define", keyDefinitions);
+                }, "/data_source/key_define/type/" + dataSourceTypeId + "/" + scopeValue,
+                "Fail to get key definitions of data source type[查询数据源参数键值对失败]");
+    }
+*/
+
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java:116]: Please help delete the comment code

File: linkis-public-enhancements/linkis-datasource/datasource-client/pom.xml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>linkis-public-enhancements</artifactId>
+        <groupId>com.webank.wedatasphere.linkis</groupId>
+        <version>1.0.0-RC1</version>
```
[the remark this answers is not in the document]
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/datasource-client/pom.xml:23]: Thanks.

File: linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java [hunk 7]
```diff
@@ -64,227 +67,415 @@
...
+            List<DataSourceParamKeyDefinition> keyDefinitionList = dataSourceRelateService
+                    .getKeyDefinitionsByType(dataSource.getDataSourceTypeId());
+            parameterValidator.validate(keyDefinitionList, connectParams);
+            //Encrypt password value type
+            RestfulApiHelper.encryptPasswordKey(keyDefinitionList, connectParams);
+
+            long versionId = dataSourceInfoService.insertDataSourceParameter(keyDefinitionList, datasourceId, connectParams, userName, comment);
+
+            return Message.ok().data("version", versionId);
+        }, "/data_source/parameter/" + datasourceId + "/json", "Fail to insert data source parameter [保存数据源参数失败]");
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java:229]: Please help delete the comment code

File: linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java [hunk 8]
```diff
@@ -64,227 +67,415 @@
...
                 return Message.error("Fail to update data source[更新数据源失败], " +
-                        "[Please check the id:'" + dataSourceId +"' and create system: '"+dataSource.getCreateSystem()+" is correct ']");
+                        "[Please check the id:'" + dataSourceId + "' and create system: '" + dataSource.getCreateSystem() + " is correct ']");
             }
             dataSource.setCreateUser(storedDataSource.getCreateUser());
-            updateDataSourceConfig(dataSource, storedDataSource);
+            dataSourceInfoService.updateDataSourceInfo(dataSource);
+//            updateDataSourceConfig(dataSource);
             return Message.ok().data("update_id", dataSourceId);
         }, "/data_source/info/" + dataSourceId + "/form", "Fail to update data source[更新数据源失败]");
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java:428]: Please help delete the comment code

File: linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/service/impl/DataSourceInfoServiceImpl.java [hunk 9]
```diff
@@ -62,62 +61,125 @@
...
+//                            && connectParams.containsKey(keyDefinition.getKey())) {
+//                        try {
+//                            //Proxy creator to delete resource
+//                            bmlAppService.clientRemoveResource(dataSource.getCreateUser(), String
+//                                    .valueOf(connectParams.get(keyDefinition.getKey())));
+//                        } catch (Exception e) {
+//                            //Ignore remove error
+//                        }
+//                    }
+//                });
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/service/impl/DataSourceInfoServiceImpl.java:182]: Please help delete the comment code

File: linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/conf/application.yml [hunk 10]
```diff
@@ -0,0 +1,23 @@
+server:
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/conf/application.yml:1]: This file is not neccessary in here.

File: linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/conf/linkis.properties [hunk 11]
```diff
@@ -0,0 +1,19 @@
+#
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/conf/linkis.properties:1]: This file is not neccessary in here.

File: linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/src/main/assembly/distribution.xml [hunk 12]
```diff
@@ -0,0 +1,206 @@
+<!--
```
Comment (wushengyeyouya) [inline linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/src/main/assembly/distribution.xml:1]: This file is not neccessary in here.

File: linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala (hunk 2 above)
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala]: Thanks，I will delete it

File: linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/conf/linkis.properties (hunk 11 above)
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/metadatamanager/service/kafka/conf/linkis.properties:1]: OK

File: linkis-public-enhancements/linkis-datasource/datasource-client/src/main/scala/com/webank/wedatasphere/linkis/datasource/client/AbstractRemoteClient.scala [hunk 13]
```diff
@@ -0,0 +1,15 @@
+package com.webank.wedatasphere.linkis.datasource.client
```
[the remark this answers is not in the document]
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/datasource-client/src/main/scala/com/webank/wedatasphere/linkis/datasource/client/AbstractRemoteClient.scala:1]: OK

File: linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala (hunk 3 above)
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/datasource-client/src/test/scala/com/webank/wedatasphere/linkis/datasource/client/TestDataSourceClient.scala:13]: ok

File: linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java (hunk 5 above)
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java:116]: OK

File: linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java (hunk 7 above)
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/restful/DataSourceCoreRestfulApi.java:229]: OK

File: linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/service/impl/DataSourceInfoServiceImpl.java (hunk 9 above)
Comment (Adamyuanyuan) [reply linkis-public-enhancements/linkis-datasource/datasourcemanager/server/src/main/java/com/webank/wedatasphere/linkis/datasourcemanager/core/service/impl/DataSourceInfoServiceImpl.java:182]: ok

**Pull request closed without merging**